src/components/Button.js

```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import Proptypes from "prop-types";
+
+function Button(props) {
+    const { type = "default", children, icon, ...rest } = props;
+
+    return (
+        <button type={type} className="btn" {...rest}>
+            {icon && <i className={`fa fa-${icon}`}></i>}&nbsp;
+            {children}
+        </button>
+    );
+}
+
+Button.propTypes = {
+    children: Proptypes.any,
+    type: Proptypes.string,
+    icon: Proptypes.string,
```

Comment:
Since all the Props are mandatory, you can include .isRequired too

Reply:
icon is not required, same with type... @gautism1 default type should be `button` not `default`

valid types `<button type="button|submit|reset">` 

make sure you understand HTML element's attribute what value it support before implement